Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeCollection.cpp

```diff
@@ -210,6 +212,15 @@ namespace AzToolsFramework
             return InComponentMode() ? m_activeComponentTypes : AZStd::vector<AZ::Uuid>{};
         }
 
+        void ComponentModeCollection::EnumerateActiveComponents(
+            AZStd::function<void(const AZ::EntityComponentIdPair&, const AZ::Uuid&)> handler) const
+        {
+            for (size_t i = 0; i < m_activeComponentModes.size(); ++i)
+            {
+                handler(m_activeComponentModes[i].first, m_activeComponentModes[i].second);
+            }
```

Comment:
question: Is there any reason why we can't use a range based for loop here?

Reply:
No, I had a different implementation before but I ended up refactoring it to support multi-entity editing. Will fix.

Reply:
Okay nice thanks